src/appsignal/client.py

```diff
@@ -48,6 +51,21 @@ def start(self) -> None:
         else:
             logger.info("AppSignal not starting: no active config found")
 
+    def stop_agent_processes(self) -> None:
+        logger.info("Killing all AppSignal agent processes")
+
+        processes = subprocess.Popen(["ps", "-A"], stdout=subprocess.PIPE)
```

Comment:
This could kill agents from other apps running on the same host. It would be nice if we look up the PID in the agent's working directory for this app. In the working directory an `agent.lock` file exists that has a format like so: `config-digest;state;PID`. If you read that file, split by `;` and take the last value you have the PID for the agent started by this app.